clients/src/test/java/org/apache/kafka/clients/producer/internals/TransactionManagerTest.java

```diff
@@ -2151,59 +2004,98 @@ public void shouldFailAbortIfAddOffsetsFailsWithFatalError() {
 
     @Test
     public void testSendOffsetsWithGroupMetadata() {
-        final long pid = 13131L;
-        final short epoch = 1;
+        Map<TopicPartition, Errors> txnOffsetCommitResponse = new HashMap<>();
+        txnOffsetCommitResponse.put(tp0, Errors.NONE);
+        txnOffsetCommitResponse.put(tp1, Errors.COORDINATOR_LOAD_IN_PROGRESS);
 
-        doInitTransactions(pid, epoch);
+        TransactionalRequestResult addOffsetsResult = prepareGroupMetadataCommit(
```

Comment:
Refactor out the common `prepareGroupMetadataCommit` to simplify the test construction